pymeasure/instruments/agilent/agilentN9320A.py

```diff
@@ -0,0 +1,262 @@
+        in dB. This property can be set.
+        """,
+        validator=strict_range,
+        values=ATT_LIMIT
+    )
+    res_bw = Instrument.control(
+        ":SENS:BAND:RES?", ":SENS:BAND:RES %e",
+        """ A floating point property that represents the resolution
+        bandwidth in Hz. This property can be set.
+        """,
+        validator=strict_discrete_set,
+        values=RES_LIMIT
+    )
+    vid_bw = Instrument.control(
+        ":SENS:BAND:VID?", ":SENS:BAND:VID %e",
+        """ A floating point property that represents the video
+        bandwidth in Hz. This property can be set.
+        """,
+        validator=strict_discrete_set,
+        values=VID_LIMIT
+    )
+
+    def __init__(self, resourceName, **kwargs):
+        super(AgilentN9320A, self).__init__(
+            resourceName,
+            "Agilent AgilentN9320A Spectrum Analyzer",
+            **kwargs
+        )
+    def init_imm(self):
+        self.write("INIT:IMM")
+
+    def init_cont(self):
+        self.write("INIT:CONT 1")
+
+    def init_single(self):
+        self.write("INIT:CONT 0")
+
+    def opc(self):
+        return int(self.ask("*OPC?"))
+
+    def avg_set(self, value=10):
+        self.write("SENS:AVER:COUN %d" % strict_range(value, [1, 1000]))
+
+    def avg(self, command='ON'):
+        self.write("SENS:AVER:STAT %s" % strict_discrete_set(command,
+                                                             ['ON', 'OFF']))
+
+    def set_peak_th(self, number=1, threshold=PEAK_TH, excursion=PEAK_EXC):
+        """Peak search mehtod using threshold and excursion"""
+        self.write("CALC:MARK:PEAK:SEAR:MODE PAR")
+        self.write("CALC:MARK:PEAK:THR:EXC %.2f" % excursion)
+        self.write("CALC:MARK%s:PEAK:THR %.2f" % (number, threshold))
+        self.write("CALC:MARK:PEAK:THR:STATE 1")
+
+    def set_peak_max(self):
+        """Peak search method: the max value of the trace is taken"""
+        self.write("CALC:MARK:PEAK:SEAR:MODE MAX")
+
+    def max(self, number=1):
+        """Peak search"""
+        self.write("CALC:MARK%s:MAX" % number)
+
+    def max_x(self, number=1):
+        """Returns the frequency in Hz at marker position"""
+        return float(self.ask("CALC:MARK%s:X?" % number))
+
+    def max_y(self, number=1):
+        """Returns the amplitude in dBm at marker position"""
+        return format(float(self.ask("CALC:MARK%s:Y?" % number)), '.4f')
+
+    def peak(self, number=1, avg=1, center=False, lr=False):
+        """ Returns the frequency and the intensity of the highest peak.
+        It can center the central frequency to the central peak.
+        It can also look the first peaks at both left and right
+        around the highest one.
+        """
+        self.avg('ON')
+        self.avg_set(avg)
+        sleep(self.DELAY)
+        self.init_single()
+        self.init_imm()
+
+        print('Averaging for %.1f seconds' % (avg*self.sweep_time))
```

Comment:
Print statements should not be used in the PyMeasure library. Instead, use a Python logger. Here is an example from another instrument:
https://github.com/ralph-group/pymeasure/blob/2dccdb4091a6a66f9d254379866bc6bd13d7a14a/pymeasure/instruments/anritsu/anritsuMS9710C.py#L36-L37
https://github.com/ralph-group/pymeasure/blob/2dccdb4091a6a66f9d254379866bc6bd13d7a14a/pymeasure/instruments/anritsu/anritsuMS9710C.py#L297-L303
Notice that `log.debug` is used for high-volume messages that typically won't be displayed.